app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -91,47 +78,99 @@ protected void onCreate(Bundle savedInstanceState) {
             startActivity(intent);
         });
 
-
         // Click listener for the add folder button
         addButton.setOnClickListener(v -> {
             // Show the create item dialog
             showCreateItemDialog();
         });
 
-        // Fill folder view (Recycler View)
         folderRecyclerView.setLayoutManager(new GridLayoutManager(this, 4));
-        folderAdapter = new FolderAdapter(
-                new FolderAdapter.ClickListener() {
-                    @Override
-                    public void onItemClick(int position, Folder model) {
-                        // Send to that folder's view
-                        PosteApplication.setSelectedFolder(model);
-                        Intent intent = new Intent(DashboardActivity.this, FolderViewActivity.class );
-                        startActivity(intent);
-                    }
 
-                    @Override
-                    public void onItemLongClick(int position, Folder model) {
-                        // Prompt to share folder
-                        Toast.makeText(DashboardActivity.this, getString(R.string.share_folder), Toast.LENGTH_LONG).show();
-                    }
-                },
-                new ArrayList<>(userFolders)
-        );
-        folderRecyclerView.setAdapter(folderAdapter);
-        registerForContextMenu(folderRecyclerView);
+        // This callback defines what will happen after the user's folders and posts are updated
+        // It is a way to handle asynchronous code.
+        // Since the updateFoldersAndPosts call happens off the main thread, we cannot expect
+        // that it will finish in time for us to use the updated data in the onCreate method.
+        // Instead, we define a callback that will be called when the update is finished.
+        // This callback is passed to the updateFoldersAndPosts method, which will call it
+        // when the update is finished saying it succeeded or failed.
+        // We simply define what we will do if it succeeds or fails.
+        UpdateCallback updateCallback = new UpdateCallback() {
+            /**
+             * This defines what will happen if the update succeeds; what we want the application
+             * to do in a success case.
+             * Again, this is assuming that a User.getUser().updateFoldersAndPosts() call was made
+             * and that the callback was passed to it; so, what do we want to do if the update
+             * was successful?
+             */
+            @Override
+            public void onSuccess() {
+                userFolders = currentUser.getFolders(); // update our local copy of the folders
+                folderNameToIdMap.clear(); // clear our map of folder names to folder IDs
+                for (Folder folder : userFolders) {  // repopulate the map
+                    folderNameToIdMap.put(
+                            folder.getTitle(),
```

Comment:
Pretty sure you would want the ID as the key here, and Title/Folder as the mapped value

Reply:
I only moved (after renaming) the mapping; can you elaborate? 

This populates the map with title as key and id as values, and later populates the folder names by key and identifies them by their id. Maybe there's something I'm missing?

Reply:
The title may or may not be unique. So if 2 folders have the same title, the second would overwrite the first and the app would only show one folder.

Reply:
Oh, you right. Good looking out. It was originally created as a Hashmap (don't recall who created it) and I just changed the name) but was only considering it as a list. Makes sense. ID to Title makes perfect sense there, then.

Reply:
Actually, I just realized that I abstracted out the usage of the map altogether -- that's why I didn't notice earlier. Good eye though! I will be removing this completely though as it is no longer used.